pkg/android/phoenix/src/com/retroarch/browser/retroactivity/RetroActivityFuture.java

```diff
@@ -1,71 +1,87 @@
+    public void handleMessage(Message msg) {
+      boolean state = (msg.arg1 == HANDLER_ARG_TRUE) ? true : false;
+
+      if (msg.what == HANDLER_WHAT_TOGGLE_IMMERSIVE) {
+        attemptToggleImmersiveMode(state);
+      } else if (msg.what == HANDLER_WHAT_TOGGLE_POINTER_CAPTURE) {
+        attemptTogglePointerCapture(state);
+      } else if (msg.what == HANDLER_WHAT_TOGGLE_POINTER_NVIDIA) {
+        attemptToggleNvidiaCursorVisibility(state);
+      } else if (msg.what == HANDLER_WHAT_TOGGLE_POINTER_ICON) {
+        attemptTogglePointerIcon(state);
+      }
+    }
+  };
+
+  @Override
+  public void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+
+    mDecorView = getWindow().getDecorView();
+
+    // If QUITFOCUS parameter is provided then enable that Retroarch quits when focus is lost
+    quitfocus = getIntent().hasExtra("QUITFOCUS");
+  }
+
   @Override
   public void onResume() {
     super.onResume();
 
     setSustainedPerformanceMode(sustainedPerformanceMode);
 
-    if (Build.VERSION.SDK_INT >= 19) {
-      // Immersive mode
-
-      // Constants from API > 14
-      final int API_SYSTEM_UI_FLAG_LAYOUT_STABLE = 0x00000100;
-      final int API_SYSTEM_UI_FLAG_LAYOUT_HIDE_NAVIGATION = 0x00000200;
-      final int API_SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN = 0x00000400;
-      final int API_SYSTEM_UI_FLAG_FULLSCREEN = 0x00000004;
-      final int API_SYSTEM_UI_FLAG_IMMERSIVE_STICKY = 0x00001000;
-
-      View thisView = getWindow().getDecorView();
-      thisView.setSystemUiVisibility(API_SYSTEM_UI_FLAG_LAYOUT_STABLE
-          | API_SYSTEM_UI_FLAG_LAYOUT_HIDE_NAVIGATION
-          | API_SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN
-          | View.SYSTEM_UI_FLAG_HIDE_NAVIGATION
-          | API_SYSTEM_UI_FLAG_FULLSCREEN
-          | API_SYSTEM_UI_FLAG_IMMERSIVE_STICKY);
-
-      // Check for Android UI specific parameters
-      Intent retro = getIntent();
-
-      if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.LOLLIPOP) {
-        String refresh = retro.getStringExtra("REFRESH");
-
-        // If REFRESH parameter is provided then try to set refreshrate accordingly
-        if(refresh != null) {
-          WindowManager.LayoutParams params = getWindow().getAttributes();
-          params.preferredRefreshRate = Integer.parseInt(refresh);
-          getWindow().setAttributes(params);
-        }
-      }
-
-      // If QUITFOCUS parameter is provided then enable that Retroarch quits when focus is lost
-      quitfocus = retro.hasExtra("QUITFOCUS");
+    // Check for Android UI specific parameters
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+      String refresh = getIntent().getStringExtra("REFRESH");
 
-      // If HIDEMOUSE parameters is provided then hide the mourse cursor
-      // This requires NVIDIA Android extensions (available on NVIDIA Shield), if they are not
-      // available then nothing will be done
-      if (retro.hasExtra("HIDEMOUSE")) hideMouseCursor();
```

Comment:
I forgot to mention - I removed this because it doesn't seem like it's being used anywhere, and it's not referenced anywhere in documentation or anything. This PR uses the `input_auto_mouse_grab` to determine if it should hide/capture the mouse, which I think is better anyway. Happy to restore it if anyone disagrees, though